src/uct/tcp/tcp.h

```diff
@@ -195,8 +195,8 @@ enum {
 typedef struct uct_tcp_cm_conn_req_pkt {
     uct_tcp_cm_conn_event_t event;      /* Connection event ID */
     uint8_t                 flags;      /* Packet flags */
-    struct sockaddr_storage iface_addr; /* Socket address of UCT local iface */
     uct_tcp_ep_cm_id_t      cm_id;      /* EP connection mananger ID */
+    struct sockaddr_storage iface_addr; /* Socket address of UCT local iface */
```

Comment:
the packet size should be according to actual address size, no need to send too much data
@dmitrygx WDYT?

Reply:
do you mean it will break wire compatibility?

Reply:
it will break, but will not affect client/server connection right?
i meant that maybe better not include iface_addr in this struct, but instead add it after when packing, depending on actual size of address family

Reply:
> i meant that maybe better not include iface_addr in this struct, but instead add it after when packing, depending on actual size of address family

makes sense to me

Reply:
> it will break, but will not affect client/server connection right?
we send CONN_REQ packet in both CONNECT_TO_EP/CONNECT_TO_IFACE cases

Reply:
packed iface_addr based on size.  please review.

Reply:
i;m seeing the following build error. can you suggest a way to handle this?

```
/__w/2/s/src/uct/tcp/tcp_cm.c:244:45: error: taking address of packed member of 'struct uct_tcp_cm_conn_req_pkt' may result in an unaligned pointer value [-Werror=address-of-packed-member] <--[gcc]
  244 |         ucs_sockaddr_copy((struct sockaddr*)conn_pkt->iface_addr,

```